packages/plugins/src/foundry/utils/index.ts

```diff
@@ -448,6 +420,14 @@ export const makeGetConfigArtifacts = (
         name: fileName,
         time: statSync(path.join(buildInfoPath, fileName)).mtime.getTime(),
       }))
+      // Sort the build info files according to their `time` field, which equals the `mtime`, i.e.
+      // the time that the build info was last modified. We'll use this field to determine the order
+      // in which to search the cache for the build info file and contract artifact that correspond
+      // to each init code supplied to this function. It's possible for the user to modify their
+      // build info files, which will change the `mtime` of the file. However, this is acceptable
+      // because we search for the build info file and contract artifact using an exact match on the
+      // init code. This means the worst case is that we select a valid yet outdated build info and
+      // artifact.
```

Comment:
Closes this ticket: [Document why it's safe to use mtime in makeGetConfigArtifacts](https://linear.app/chugsplash/issue/CHU-751/document-why-its-safe-to-use-mtime-in-makegetconfigartifacts)